plugin/vscode/test/vscode-tests.js

```diff
@@ -1095,6 +1095,196 @@ tests = {
+    let helloFilePath = path.join(scratchDirectory, "hello.js");
+    fs.writeFileSync(helloFilePath, "let x = undeclaredVariable;\n");
+    let helloURI = vscode.Uri.file(helloFilePath);
+    let helloDocument = await vscode.workspace.openTextDocument(helloURI);
+    await loadExtensionAsync({ addCleanup });
+    let helloEditor = await vscode.window.showTextDocument(helloDocument);
+
+    // 1. Make sure we're snarky at the start
+    await waitUntilAnyDiagnosticsAsync(helloURI);
+    let diags = normalizeDiagnostics(helloURI);
+    diags = diags.map(({ code, severity, message }) => ({
+      code,
+      severity,
+      message,
+    }));
+    assert.deepStrictEqual(diags, [
+      // use of undeclared variable 'undeclaredVariable'
+      {
+        code: {
+          target: "https://quick-lint-js.com/errors/E0057/",
+          value: "E0057",
+        },
+        severity: vscode.DiagnosticSeverity.Warning,
+        message: "did you fail spelling class?",
+      },
+    ]);
+
+    // 2. Disable snarky
+    await vscode.workspace
+      .getConfiguration("quick-lint-js")
+      .update("snarky", false, vscode.ConfigurationTarget.Workspace);
+
+    // 3. Make sure we're polite now
+    await pollAsync(() => {
+      let diags = normalizeDiagnostics(helloURI);
+      diags = diags.map(({ code, severity, message }) => ({
+        code,
+        severity,
+        message,
+      }));
+      let got = diags;
+      let want = [
+        {
+          code: {
+            target: "https://quick-lint-js.com/errors/E0057/",
+            value: "E0057",
+          },
+          severity: vscode.DiagnosticSeverity.Warning,
+          message: "use of undeclared variable: undeclaredVariable",
+        },
+      ];
+      assert.deepStrictEqual(diags, want);
+    });
+  },
+  snarky_enabled_at_start_config_file: async ({ addCleanup }) => {
+    addCleanup(resetConfigurationAsync);
+    await vscode.workspace
+      .getConfiguration("quick-lint-js")
+      .update("snarky", true, vscode.ConfigurationTarget.Workspace);
+    let scratchDirectory = makeScratchDirectory({ addCleanup });
+    let configFilePath = path.join(scratchDirectory, "quick-lint-js.config");
+    fs.writeFileSync(configFilePath, "{");
+    let configURI = vscode.Uri.file(configFilePath);
+
+    await loadExtensionAsync({ addCleanup });
+    let configDocument = await vscode.workspace.openTextDocument(configURI);
+    let configEditor = await vscode.window.showTextDocument(configDocument);
+
+    await waitUntilAnyDiagnosticsAsync(configURI);
+
+    let configDiags = normalizeDiagnostics(configURI);
+    assert.deepStrictEqual(
+      configDiags.map(({ code, message }) => ({ code, message })),
+      [
+        {
+          code: {
+            target: "https://quick-lint-js.com/errors/E0164/",
+            value: "E0164",
+          },
+          message: "yeah, JSON sucks; try quick-lint-json",
+        },
+      ]
+    );
+  },
```

Comment:
I should probably add `enable_snarky_config_file` and `disable_snarky_config_file` tests

Reply:
I think you can add quick-lint-js.config stuff into the other three tests.

Alternatively, you could data-drive the tests (see line 24 for an existing example):
```javascript
for (let testCase of [
  {
    fileName: 'hello.js',
    content: 'undeclaredVariable',
    englishMessage: 'use of undeclared variable: undeclaredVariable',
    snarkyEnglishMessage: 'did you fail spelling class?',
  },
  {
    fileName: 'quick-lint-js.config',
    content: '{',
    englishMessage: '...',
    snarkyEnglishMessage: 'yeah, JSON sucks; try quick-lint-json',
  },
]) {
  tests = {
    ...tests,
    [`enabling snarky re-lints (${testCase.fileName})`]: async ({ addCleanup }) => {
      ...
    },
  };
}
```

Reply:
That's an awesome idea 💡.  This is why they pay you the big bucks